app/pre_oredr.py

```diff
@@ -0,0 +1,7 @@
+def preorder_traversal(root):
```

Comment:
**Code optimization Suggestion**
```python
def preorder_traversal(root):
    result = []
    stack = [root]
    while stack:
        node = stack.pop()
        if node:
            result.append(node.value)
            stack.append(node.right)
            stack.append(node.left)
    return result
```
**Explanation for suggestion**
The original recursive approach is refactored to an iterative approach using a stack. This avoids potential stack overflow for deep trees and may improve performance due to reduced function call overhead.